src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,522 @@
+        The VSS specification doesn't provide any types that can be translated into ApiGear operations,
+        (which are functions that may be executed on the interface), or ApiGear signals (which are more like events).
+        """
+
+
+class ApiGearModule:
+    def __init__(self):
+        super().__init__()
+
+        self.interfaces: typing.List[ApiGearInterface] = []
+        self.enumerations: typing.List[ApiGearEnumeration] = []
+        self.structures: typing.List[ApiGearStructure] = []
+
+
+def starts_with_number(name: str) -> bool:
+    return name[0].isdigit()
+
+
+def is_floating_point(datatype: VSSDataType | None) -> bool:
+    if datatype is None:
+        return False
+    return datatype in [VSSDataType.FLOAT, VSSDataType.DOUBLE]
+
+
+def hex_to_dec(number: str) -> int | None:
+    if starts_with_number(number):
+        try:
+            return int(number, 16)
+        except ValueError:
+            return None
+    return None
+
+
+def get_allowed_enum_literal(name: str) -> str:
+    """
+    Check if this is is an allowed literal name, if not add prefix.
+
+    Background:
+
+    In VSS '123' is a perfectly fine string literal, usable as allowed value for a string.
+    The current exporter (this file) translated it previously to 123 which is not a valid DSS IDL literal.
+    Adding an underscore as prefix makes the generated IDL ok, but then gives problems if generating for example
+    Python code by Eclipse Cyclone DDS idlc Python Backend.
+    By that reason we now add a regular character instead.
+    """
+    if starts_with_number(name):
+        return "d" + name.replace(".", "_")
+    return name
+
+
+def get_apigear_datatype(t: str) -> typing.Optional[ApiGearType]:
+    is_array = False
+    apigear_type = None
+    if t.endswith("[]"):
+        is_array = True
+        t = t[0:-2]
+
+    if t in ["int8", "int16", "int32", "uint8", "uint16", "uint32"]:
+        apigear_type = ApiGearType(ApiGearBasicType.INT32.value)
+    elif t in ["int64", "uint64"]:
+        apigear_type = ApiGearType(ApiGearBasicType.INT64.value)
+    elif t == "float":
+        apigear_type = ApiGearType(ApiGearBasicType.FLOAT32.value)
+    elif t == "double":
+        apigear_type = ApiGearType(ApiGearBasicType.FLOAT64.value)
+    elif t == "boolean":
+        apigear_type = ApiGearType(ApiGearBasicType.BOOL.value)
+    elif t == "string":
+        apigear_type = ApiGearType(ApiGearBasicType.STRING.value)
+    else:
+        apigear_type = ApiGearType(t.replace(".", "_"))
+
+    if is_array and apigear_type is not None:
+        apigear_type.is_array = True
+
+    return apigear_type
+
+
+def node_name(node: VSSNode) -> str:
+    return node.qualified_name().replace(".", "_")
+
+
+def has_defined_subset_of_values(node: VSSNode) -> bool:
+    return node.allowed != ""
+
+
+def generate_property(node: VSSNode, datatype: str) -> ApiGearProperty | None:
+    type = get_apigear_datatype(datatype)
```

Comment:
`type` shadows a keyword. Usually that can lead to nasty bugs

Reply:
Changed the `type` name to `apigear_type`.